packages/runtime/src/next/api-handler/handlers/proxy-preview-mode.ts

```diff
@@ -85,71 +91,63 @@ async function proxyPreviewModeApiRouteHandler(
     } catch {}
   }
 
-  previewModeProxy.once('proxyReq', proxyReq => {
-    proxyReq.removeHeader('X-Makeswift-Preview-Mode')
-
-    // The following headers are Next.js-specific and are removed to prevent Next.js from
-    // short-circuiting requests, breaking routing.
-    proxyReq.removeHeader('X-Invoke-Path')
-    proxyReq.removeHeader('X-Invoke-Query')
-
-    const url = new URL(proxyReq.path, 'http://n')
-
-    url.searchParams.delete('x-makeswift-preview-mode')
+  const {
+    'X-Makeswift-Preview-Mode': xMakeswiftPreviewMode,
+    'X-Invoke-Path': xInvokePath,
+    'X-Invoke-Query': xInvokeQuery,
+     ...headers
+  } = req.headers
 
-    proxyReq.path = url.pathname + url.search
-  })
-
-  if (req.query.secret !== apiKey) return res.status(401).send('Unauthorized')
+  const isForwardedProtoHttps = match(req.headers['X-Forwarded-Proto'])
+    .with(P.string, x => x.split(','))
+    .with(P.array(P.string), x => x)
+    .otherwise(() => [])
+    .includes('https')
 
-  const host = req.headers.host
-  const originalCookies = req.headers.cookie
+  const isForwardedSSL = match(req.headers['X-Forwarded-SSL'])
+    .with('on', () => true)
+    .otherwise(() => false)
 
-  if (host == null) return res.status(400).send('Bad Request')
-
-  const forwardedProtoHeader = req.headers['x-forwarded-proto']
-
-  let forwardedProto: string[] = []
-  if (Array.isArray(forwardedProtoHeader)) {
-    forwardedProto = forwardedProtoHeader
-  } else if (typeof forwardedProtoHeader === 'string') {
-    forwardedProto = forwardedProtoHeader.split(',')
-  }
-
-  const isForwardedProtoHttps = forwardedProto.includes('https')
+  const proto = isForwardedProtoHttps || isForwardedSSL ? 'https' : 'http'
+  const url = new URL(`${proto}://${req.headers.host}${req.url}`)
 
-  const forwardedSSL = req.headers['x-forwarded-ssl']
-  const isForwardedSSL = typeof forwardedSSL === 'string' && forwardedSSL === 'on'
+  url.searchParams.delete('x-makeswift-preview-mode')
 
-  const proto = isForwardedProtoHttps || isForwardedSSL ? 'https' : 'http'
-  let target = `${proto}://${host}`
-
-  // If the user generates a locally-trusted CA for their SSL cert, it's likely that Node.js won't
-  // trust this CA unless they used the `NODE_EXTRA_CA_CERTS` option
-  // (see https://stackoverflow.com/a/68135600). To provide a better developer experience, instead
-  // of requiring the user to provide the CA to Node.js, we don't enforce SSL during local
-  // development.
-  const secure = process.env['NODE_ENV'] === 'production'
```

Comment:
Did we check that this is still being handled properly?

Reply:
Pretty sure we don't need to now that we are just fetching the page directly. I could be wrong though 🤷‍♂️

Reply:
I think you're right since this is more of a configuration for `http-proxy` 👍🏻